SampleApp/src/PortAudioMicrophoneWrapper.cpp

```diff
@@ -128,7 +164,40 @@ bool PortAudioMicrophoneWrapper::initialize() {
     return true;
 }
 
+void PortAudioMicrophoneWrapper::ReaderThread(PortAudioMicrophoneWrapper *wrapper)
+{
+    ACSDK_LOG(alexaClientSDK::avsCommon::utils::logger::Level::INFO, alexaClientSDK::avsCommon::utils::logger::LogEntry("FileInput", "threadAlive"));
+    signed short block[128];
+    const unsigned period = sizeof(block) / 2 * 1000 / (unsigned)SAMPLE_RATE;
+    while (wrapper->m_threadFuture->wait_for(std::chrono::milliseconds(period)) == std::future_status::timeout) {
+        wrapper->m_fileStream->read((char*)block, sizeof(block));
+        if (wrapper->m_fileStream->fail() || wrapper->m_fileStream->eof()) {
+            if (!wrapper->m_eofReached) {
+                ACSDK_LOG(alexaClientSDK::avsCommon::utils::logger::Level::INFO, alexaClientSDK::avsCommon::utils::logger::LogEntry("FileInput", "eofReached"));
+                wrapper->m_eofReached = true;
+            }
+            std::memset(block, 0, sizeof(block));
+        }
+        ssize_t returnCode = wrapper->m_writer->write(block, sizeof(block) / 2);
+        if (returnCode <= 0) {
+            ACSDK_CRITICAL(LX("Failed to write to stream."));
+        }
+        if (wrapper->m_samplesRead % (unsigned)SAMPLE_RATE == 0) {
+            ACSDK_LOG(alexaClientSDK::avsCommon::utils::logger::Level::INFO, alexaClientSDK::avsCommon::utils::logger::LogEntry("FileInput", "timeElapsedApprox").d("seconds", wrapper->m_samplesRead / (unsigned)SAMPLE_RATE));
+        }
+        wrapper->m_samplesRead += sizeof(block) / 2;
+    }
+    ACSDK_LOG(alexaClientSDK::avsCommon::utils::logger::Level::INFO, alexaClientSDK::avsCommon::utils::logger::LogEntry("FileInput", "threadShuttingDown"));
+}
+
 bool PortAudioMicrophoneWrapper::startStreamingMicrophoneData() {
+    m_threadPromise = new std::promise<void>;
+    m_threadFuture = new std::future<void>{m_threadPromise->get_future()};
+    m_readerThread = new std::thread{ReaderThread, this};
+    m_samplesRead = 0;
+    m_eofReached = false;
+    ACSDK_LOG(alexaClientSDK::avsCommon::utils::logger::Level::INFO, alexaClientSDK::avsCommon::utils::logger::LogEntry("FileInput", "threadCreated"));
+    return true;
```

Comment:
Am I reading this correctly? If we just unconditionally return here, what's the point of the rest of the lines in this function? Doesn't the compiler warn about unreachable code?